store/tikv/region_cache.go

```diff
@@ -532,40 +557,97 @@ func (c *RegionCache) loadStoreAddr(bo *Backoffer, id uint64) (string, error) {
 	}
 }
 
-// DropStoreOnSendRequestFail is used for clearing cache when a tikv server does not respond.
-func (c *RegionCache) DropStoreOnSendRequestFail(ctx *RPCContext, err error) {
-	// We need to drop the store only when the request is the first one failed on this store.
-	// Because too many concurrently requests trying to drop the store will be blocked on the lock.
-	failedRegionID := ctx.Region
+// trySwitchNextPeer give up unreachable peer and tries to select another valid peer.
+// It returns false if all peers are unreachable.
+func (r *Region) trySwitchNextPeer(failedStoreID uint64) (hasReachable bool) {
+	// try other peers in rand order(set).
+	if len(r.backupPeers) > 0 {
+		var (
+			id   uint64
+			peer *metapb.Peer
+		)
+		for id, peer = range r.backupPeers {
```

Comment:
I don't think it is good practice to depend on the fact that access of map is randomized... Maybe simpler to use a slice to store backupPeers instead...